content/courses/advanced-ada/parts/data_types/types_representation.rst

```diff
@@ -2754,6 +2761,402 @@ these array declarations are equivalent:
       Add to previous section!
 
 
+.. _Adv_Ada_Addresses:
+
+Addresses
+---------
+
+In other languages, such as C, the concept of pointers and addresses plays
+a prominent role. (In fact, in C, many optimizations rely on the usage of
+pointer arithmetic.) The concept of addresses does exist in Ada, but it's
+mainly reserved for very specific applications, mostly related to low-level
+programming. In general, other approaches |mdash| such as using access types
+|mdash| are more than sufficient. (We discuss
+:doc:`access types <../resource_management/access_types>` in another chapter.
+Also, later on in that chapter, we discuss the
+:ref:`relation between access types and addresses <Adv_Ada_Access_Address>`.)
+In this section, we discuss some details about using addresses in Ada.
+
+We make use of the :ada:`Address` type, which is defined in :ada:`System`
+package, to handle addresses. In contrast to other programming languages (such
+as C or C++), an address in Ada isn't an integer value |mdash| its definition
+actually depends on the compiler implementation, but for now, let's consider
```

Comment:
It is true that "its definition actually depends on the compiler implementation" but that is driven directly by the hardware.

Reply:
Thanks for the hint! I've just added this information to the paragraph.